File: LeetCode/167.Two_Sum_II_-_Input_Array_Is_Sorted/Mario/167.Two_Sum_II_-_Input_Array_Is_Sorted.java
class Solution {
    public int[] twoSum(int[] numbers, int target) {
        int n = numbers.length;
        int i = 0, j = n - 1, mid = 0;
        while (i < j) {
            mid = (i + j) / 2;
            int sum = numbers[i] + numbers[j];
            if (sum == target) {
                return new int[] { i + 1, j + 1 };
            } else if (sum > target) {
                if (numbers[i] + numbers[mid] > target) {
                    j = mid - 1;
                } else {
                    j -= 1;
                }
            } else {
                if (numbers[j] + numbers[mid] < target) {
                    i = mid + 1;
                } else {
                    i += 1;
                }
            }
        }

        return new int[] {};
    }
}

File: LeetCode/167.Two_Sum_II_-_Input_Array_Is_Sorted/167.Two_Sum_II_-_Input_Array_Is_Sorted.py
#!/usr/bin/env python3

# 167. Two Sum II - Input Array Is Sorted
class Solution:
    def twoSum(self, numbers: List[int], target: int) -> List[int]:
        i, j = 0, len(numbers)-1
        while i < j:
            ans = numbers[i] + numbers[j]
            if ans > target:
                j -= 1
            elif ans < target:
                i += 1
            else:
                return [i+1, j+1]
